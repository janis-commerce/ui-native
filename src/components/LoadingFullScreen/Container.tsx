// import React, {useState} from 'react';
// import {StyleSheet, SafeAreaView, View, TouchableHighlight, Text} from 'react-native';
// import LoadingFullScreen from './index';

// const Container = () => {
// 	const [isloading, setisloading] = useState(false);
// 	const Toogle = () => {
// 		setisloading(true);
// 		setTimeout(() => {
// 			setisloading(false);
// 		}, 3000);
// 	};

// 	return (
// 		<SafeAreaView style={styles.ContainerStyles}>
// 			<View style={styles.ContainerStyles}>
// 				<TouchableHighlight style={styles.ButonStyle} onPress={Toogle}>
// 					<Text style={styles.TextStyles}>{isloading ? 'Loading' : 'press'}</Text>
// 				</TouchableHighlight>
// 				<LoadingFullScreen visible={isloading} text="Aguarde mientras procesamos los datos.." />
// 			</View>
// 		</SafeAreaView>
// 	);
// };

// const styles = StyleSheet.create({
// 	ContainerStyles: {
// 		flex: 1,
// 		// width: 100,
// 		// height: 500,
// 		backgroundColor: 'white',
// 		justifyContent: 'center',
// 		alignItems: 'center',
// 	},
// 	ButonStyle: {
// 		width: 200,
// 		height: 50,
// 		backgroundColor: 'red',
// 		justifyContent: 'center',
// 		alignItems: 'center',
// 	},
// 	TextStyles: {
// 		color: 'white',
// 		fontSize: 24,
// 	},
// });

// export default Container;
