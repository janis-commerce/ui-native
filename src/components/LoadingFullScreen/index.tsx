import React from 'react';
import {StyleSheet, Modal, ModalProps, Text} from 'react-native';

interface ILoadingFullScreen extends ModalProps {
	text?: string;
}

const LoadingFullScreen = ({text, ...props}: ILoadingFullScreen) => {
	return (
		<Modal
			visible={true}
			transparent
			animationType="fade"
			testID="loading modal"
			style={styles.ModalStyles}
			{...props}>
			<Text style={styles.TextStyles}>{text}</Text>
		</Modal>
	);
};

const styles = StyleSheet.create({
	ModalStyles: {
		width: 100,
		height: 100,
		display: 'flex',
		flexDirection: 'row',
	},
	TextStyles: {
		fontSize: 16,
		fontFamily: 'Roboto',
	},
});

export default LoadingFullScreen;
