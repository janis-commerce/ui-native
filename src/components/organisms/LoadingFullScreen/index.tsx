import React from 'react';
import {StyleSheet, Modal, ModalProps, Text, View, ViewStyle} from 'react-native';
import Loading from 'atoms/Loading';
import Svg from 'atoms/Svg';
import {grey, white} from 'theme/palette';
import {Names} from 'ts/interfaces/svgs';
import {moderateScale, horizontalScale, scaledForDevice} from 'scale';

interface ILoadingFullScreen extends ModalProps {
	isLoading: boolean;
	text?: string;
	svgName?: Names;
	spinnerDuration?: number;
	style?: ViewStyle;
}

const validFontSize = scaledForDevice(16, moderateScale);
const validLineHeight = scaledForDevice(24, moderateScale);
const validMarginTop = scaledForDevice(25, moderateScale);
const validWidth = scaledForDevice(36, horizontalScale);
const validHeight = scaledForDevice(25, moderateScale);

const styles = StyleSheet.create({
	ContainerStyles: {
		flex: 1,
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: white.semiTransparent,
	},
	TextStyles: {
		fontSize: validFontSize,
		lineHeight: validLineHeight,
		fontFamily: 'Roboto',
		color: grey[700],
		textAlign: 'center',
		fontWeight: '500',
		width: '50%',
		marginTop: validMarginTop,
	},
});

const LoadingFullScreen = ({
	text,
	isLoading,
	svgName = 'janis-logo',
	spinnerDuration = 2000,
	style,
	...props
}: ILoadingFullScreen) => {
	const hasTextPassed = typeof text === 'string' && Boolean(text);
	return (
		<Modal visible={isLoading} transparent animationType="fade" testID="loading modal" {...props}>
			<View style={[styles.ContainerStyles, style]}>
				<Loading isLoading={isLoading} duration={spinnerDuration}>
					<Svg name={svgName} width={validWidth} height={validHeight} />
				</Loading>
				{hasTextPassed && <Text style={styles.TextStyles}>{text}</Text>}
			</View>
		</Modal>
	);
};

export default LoadingFullScreen;
