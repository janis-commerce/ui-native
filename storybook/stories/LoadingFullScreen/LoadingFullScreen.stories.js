import React, {useState} from 'react';
import {TouchableHighlight, Text, StyleSheet} from 'react-native';
import LoadingFullScreen from '../../../src/components/LoadingFullScreen';
import {primary} from '../../../src/theme/palette';
import {moderateScale} from '../../../src/utils';

export default {
	title: 'Components/LoadingFullScreen',
	argTypes: {
		text: {
			options: ['Aguarde mientras procesamos los datos...', ''],
			control: {
				type: 'select',
				labels: {'': 'Sin texto'},
			},
		},
	},
};

const styles = StyleSheet.create({
	ButtonStyle: {
		paddingLeft: moderateScale(15),
		paddingRight: moderateScale(15),
		height: moderateScale(50),
		borderRadius: moderateScale(25),
		backgroundColor: primary.main,
		justifyContent: 'center',
		alignItems: 'center',
	},
	TextStyles: {
		color: 'white',
		fontSize: moderateScale(24),
	},
});

export const DefaultProps = (props) => {
	const [isloading, setisloading] = useState(false);

	const Toogle = () => {
		setisloading(true);
		setTimeout(() => {
			setisloading(false);
		}, 3000);
	};

	return (
		<>
			<TouchableHighlight style={styles.ButtonStyle} onPress={Toogle}>
				<Text style={styles.TextStyles}>Preview loading screen</Text>
			</TouchableHighlight>
			<LoadingFullScreen {...props} isLoading={isloading} />
		</>
	);
};

DefaultProps.storyName = 'default props';

DefaultProps.args = {
	text: 'Aguarde mientras procesamos los datos...',
};
