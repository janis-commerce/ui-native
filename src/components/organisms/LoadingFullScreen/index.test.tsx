import 'react-native';
import React from 'react';
import {create} from 'react-test-renderer';
import LoadingFullScreen from './index';

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.spyOn(React, 'useEffect').mockImplementation((f) => f());

describe('LoadingFullScreen component', () => {
	describe('Modal is not visible', () => {
		it('when is visible is false', () => {
			const component = create(<LoadingFullScreen isLoading={false} />);
			const {isLoading} = component.root.props;
			expect(isLoading).toBe(false);
		});
	});

	describe('render correct', () => {
		it('render only the loading', () => {
			const {toJSON} = create(<LoadingFullScreen isLoading />);
			expect(toJSON()).toMatchSnapshot();
		});

		it('renders well the loading next to the passed text', () => {
			const {toJSON} = create(<LoadingFullScreen isLoading text="Lorem ipsum" />);
			expect(toJSON()).toMatchSnapshot();
		});
	});
});
