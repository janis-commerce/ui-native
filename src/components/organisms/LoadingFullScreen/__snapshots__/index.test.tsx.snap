// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingFullScreen component render correct render only the loading 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  testID="loading modal"
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffffbf",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 133.5,
            "justifyContent": "center",
            "position": "relative",
            "width": 133.5,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignItems": "center",
            "height": 64,
            "justifyContent": "center",
            "position": "absolute",
            "transform": Array [
              Object {
                "rotateZ": "0deg",
              },
            ],
            "width": 64,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={133.5}
          bbWidth={133.5}
          focusable={false}
          height={133.5}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 133,
                "width": 133,
              },
            ]
          }
          vbHeight={163}
          vbWidth={163}
          width={133.5}
          x="0px"
          y="0px"
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M134.1,136.4c-30.3,29.1-78.5,28-107.5-2.3c-29.1-30.3-28-78.5,2.3-107.5s78.5-28,107.5,2.3"
              fill={null}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={4291875811}
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="10.5546"
            />
            <RNSVGPath
              d="M78.4,5.5c42-1.7,77.4,30.9,79.1,72.9c1.7,42-30.9,77.4-72.9,79.1c-42,1.7-77.4-30.9-79.1-72.9"
              fill={null}
              name="path-color"
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={4280908287}
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="10.5546"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View>
        <View
          height={108.5}
          width={156.5}
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={108.5}
            bbWidth={156.5}
            fill="none"
            focusable={false}
            height={108.5}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 108,
                  "width": 156,
                },
              ]
            }
            vbHeight={56}
            vbWidth={60}
            width={156.5}
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M34.58 37.8333L31.0014 41.3821C26.0518 46.2905 18.0148 46.2905 13.0652 41.3821L3.02961 31.4301C-0.403882 28.0252 -0.403884 22.4966 3.02961 19.0917L3.23893 18.8841C6.67242 15.4792 12.2475 15.4792 15.681 18.8841L34.58 37.6257C34.6378 37.6829 34.6378 37.776 34.58 37.8333Z"
                fill={
                  Array [
                    1,
                    "paint0_linear_27636_2339",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
              <RNSVGPath
                d="M31.9219 40.6336L44.2018 52.8112C47.6319 56.2127 53.2204 56.4646 56.7701 53.1861C60.4966 49.7441 60.5749 43.9559 57.0047 40.4155L45.1625 28.672C44.9262 28.3668 44.6756 28.0691 44.3938 27.7896L37.6527 21.1046C34.8015 17.6154 35.0114 12.4834 38.2845 9.2373L32.008 15.4616L31.9624 15.4618L31.9518 15.4722L31.9413 15.4618L25.4554 21.8937C22.0485 25.2722 22.0435 30.7483 25.4443 34.1329L31.9219 40.6336Z"
                fill={
                  Array [
                    1,
                    "paint1_linear_27636_2339",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
              <RNSVGPath
                d="M37.5668 46.2315C34.324 43.0157 35.0581 37.533 38.3224 34.2864L56.9837 15.7806C60.4773 12.3161 60.4773 6.69912 56.9837 3.23461C53.4901 -0.229897 47.8259 -0.229897 44.3323 3.23461L28.3167 19.0563L25.3305 22.0175C21.9914 25.3288 21.9844 30.7101 25.3291 34.0159L31.9099 40.6262"
                fill={
                  Array [
                    1,
                    "paint2_linear_27636_2339",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -29424,
                      0.2371,
                      -33524,
                      0.5375,
                      -36599,
                      0.8395,
                      -37368,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="paint0_linear_27636_2339"
                  x1="7.89316"
                  x2="32.0739"
                  y1="23.8231"
                  y2="47.7674"
                />
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0.0004,
                      -15264001,
                      0.2994,
                      -14861057,
                      0.806,
                      -14059009,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="paint1_linear_27636_2339"
                  x1="32.8934"
                  x2="73.1246"
                  y1="22.735"
                  y2="64.4611"
                />
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -11300097,
                      0.4865,
                      -12811009,
                      0.9991,
                      -14059009,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="paint2_linear_27636_2339"
                  x1="31.5096"
                  x2="57.4251"
                  y1="34.0179"
                  y2="8.09813"
                />
              </RNSVGDefs>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`LoadingFullScreen component render correct renders well the loading next to the passed text 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  testID="loading modal"
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffffbf",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 133.5,
            "justifyContent": "center",
            "position": "relative",
            "width": 133.5,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignItems": "center",
            "height": 64,
            "justifyContent": "center",
            "position": "absolute",
            "transform": Array [
              Object {
                "rotateZ": "0deg",
              },
            ],
            "width": 64,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={133.5}
          bbWidth={133.5}
          focusable={false}
          height={133.5}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 133,
                "width": 133,
              },
            ]
          }
          vbHeight={163}
          vbWidth={163}
          width={133.5}
          x="0px"
          y="0px"
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M134.1,136.4c-30.3,29.1-78.5,28-107.5-2.3c-29.1-30.3-28-78.5,2.3-107.5s78.5-28,107.5,2.3"
              fill={null}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={4291875811}
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="10.5546"
            />
            <RNSVGPath
              d="M78.4,5.5c42-1.7,77.4,30.9,79.1,72.9c1.7,42-30.9,77.4-72.9,79.1c-42,1.7-77.4-30.9-79.1-72.9"
              fill={null}
              name="path-color"
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={4280908287}
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="10.5546"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View>
        <View
          height={108.5}
          width={156.5}
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={108.5}
            bbWidth={156.5}
            fill="none"
            focusable={false}
            height={108.5}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 108,
                  "width": 156,
                },
              ]
            }
            vbHeight={56}
            vbWidth={60}
            width={156.5}
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M34.58 37.8333L31.0014 41.3821C26.0518 46.2905 18.0148 46.2905 13.0652 41.3821L3.02961 31.4301C-0.403882 28.0252 -0.403884 22.4966 3.02961 19.0917L3.23893 18.8841C6.67242 15.4792 12.2475 15.4792 15.681 18.8841L34.58 37.6257C34.6378 37.6829 34.6378 37.776 34.58 37.8333Z"
                fill={
                  Array [
                    1,
                    "paint0_linear_27636_2339",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
              <RNSVGPath
                d="M31.9219 40.6336L44.2018 52.8112C47.6319 56.2127 53.2204 56.4646 56.7701 53.1861C60.4966 49.7441 60.5749 43.9559 57.0047 40.4155L45.1625 28.672C44.9262 28.3668 44.6756 28.0691 44.3938 27.7896L37.6527 21.1046C34.8015 17.6154 35.0114 12.4834 38.2845 9.2373L32.008 15.4616L31.9624 15.4618L31.9518 15.4722L31.9413 15.4618L25.4554 21.8937C22.0485 25.2722 22.0435 30.7483 25.4443 34.1329L31.9219 40.6336Z"
                fill={
                  Array [
                    1,
                    "paint1_linear_27636_2339",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
              <RNSVGPath
                d="M37.5668 46.2315C34.324 43.0157 35.0581 37.533 38.3224 34.2864L56.9837 15.7806C60.4773 12.3161 60.4773 6.69912 56.9837 3.23461C53.4901 -0.229897 47.8259 -0.229897 44.3323 3.23461L28.3167 19.0563L25.3305 22.0175C21.9914 25.3288 21.9844 30.7101 25.3291 34.0159L31.9099 40.6262"
                fill={
                  Array [
                    1,
                    "paint2_linear_27636_2339",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -29424,
                      0.2371,
                      -33524,
                      0.5375,
                      -36599,
                      0.8395,
                      -37368,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="paint0_linear_27636_2339"
                  x1="7.89316"
                  x2="32.0739"
                  y1="23.8231"
                  y2="47.7674"
                />
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0.0004,
                      -15264001,
                      0.2994,
                      -14861057,
                      0.806,
                      -14059009,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="paint1_linear_27636_2339"
                  x1="32.8934"
                  x2="73.1246"
                  y1="22.735"
                  y2="64.4611"
                />
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -11300097,
                      0.4865,
                      -12811009,
                      0.9991,
                      -14059009,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="paint2_linear_27636_2339"
                  x1="31.5096"
                  x2="57.4251"
                  y1="34.0179"
                  y2="8.09813"
                />
              </RNSVGDefs>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#585858",
          "fontFamily": "Roboto",
          "fontSize": 33.5,
          "fontWeight": "500",
          "lineHeight": 50,
          "marginTop": 52,
          "textAlign": "center",
          "width": "50%",
        }
      }
    >
      Lorem ipsum
    </Text>
  </View>
</Modal>
`;
